pkg/download/downloader.go

```diff
@@ -131,9 +132,12 @@ func extractTARGZ(targetDir string, at io.ReaderAt, size int64) error {
 			if err != nil {
 				return errors.Wrapf(err, "failed to create file %q", path)
 			}
+			close := func() { f.Close() }
```

Comment:
Would a `defer f.Close()` not be a cleaner solution?

Reply:
we could end up with have thousands of files open and end up exceeding open fd limit. no need to delay.